// @uniswap/analytics-events/lib/esm/interface/index.js
/**
 * Known pages in the app. Highest order context.
 */
export var PageName;
(function (PageName) {
    PageName["ABOUT_PAGE"] = "about-page";
    PageName["LANDING_PAGE"] = "landing-page";
    PageName["NFT_COLLECTION_PAGE"] = "nft-collection-page";
    PageName["NFT_DETAILS_PAGE"] = "nft-details-page";
    PageName["NFT_EXPLORE_PAGE"] = "nft-explore-page";
    PageName["NFT_PROFILE_PAGE"] = "nft-profile-page";
    PageName["POOL_PAGE"] = "pool-page";
    PageName["SWAP_PAGE"] = "swap-page";
    PageName["TOKENS_PAGE"] = "tokens-page";
    PageName["TOKEN_DETAILS_PAGE"] = "token-details";
    PageName["VOTE_PAGE"] = "vote-page";
})(PageName || (PageName = {}));
/**
 * Sections. Disambiguates low-level elements that may share a name.
 * eg a `back` button in a modal will have the same `element`,
 * but a different `section`.
 */
export var SectionName;
(function (SectionName) {
    SectionName["CURRENCY_INPUT_PANEL"] = "swap-currency-input";
    SectionName["CURRENCY_OUTPUT_PANEL"] = "swap-currency-output";
    SectionName["NAVBAR_SEARCH"] = "Navbar Search";
    SectionName["WIDGET"] = "widget";
})(SectionName || (SectionName = {}));
/** Known modals for analytics purposes. */
export var ModalName;
(function (ModalName) {
    ModalName["CONFIRM_SWAP"] = "confirm-swap-modal";
    ModalName["NFT_LISTING"] = "nft-listing-modal";
    ModalName["NFT_TX_COMPLETE"] = "nft-tx-complete-modal";
    ModalName["TOKEN_SELECTOR"] = "token-selector-modal";
})(ModalName || (ModalName = {}));
/**
 * Known element names for analytics purposes.
 * Use to identify low-level components given a TraceContext
 */
export var ElementName;
(function (ElementName) {
    ElementName["ABOUT_PAGE_ANALYTICS_CARD"] = "about-page-analytics-card";
    ElementName["ABOUT_PAGE_BUY_CRYPTO_CARD"] = "about-page-buy-crypto-card";
    ElementName["ABOUT_PAGE_DEV_DOCS_CARD"] = "about-page-dev-docs-card";
    ElementName["ABOUT_PAGE_EARN_CARD"] = "about-page-earn-card";
    ElementName["ABOUT_PAGE_NFTS_CARD"] = "about-page-nfts-card";
    ElementName["ABOUT_PAGE_SWAP_CARD"] = "about-page-swap-card";
    ElementName["AUTOROUTER_VISUALIZATION_ROW"] = "expandable-autorouter-visualization-row";
    ElementName["BLOG_LINK"] = "blog-link";
    ElementName["CAREERS_LINK"] = "careers-link";
    ElementName["COMMON_BASES_CURRENCY_BUTTON"] = "common-bases-currency-button";
    ElementName["CONFIRM_SWAP_BUTTON"] = "confirm-swap-or-send";
    ElementName["CONNECT_WALLET_BUTTON"] = "connect-wallet-button";
    ElementName["CONTINUE_BUTTON"] = "continue-button";
    ElementName["DOCS_LINK"] = "docs-link";
    ElementName["EXPLORE_BANNER"] = "explore-banner";
    ElementName["EXPLORE_SEARCH_INPUT"] = "explore_search_input";
    ElementName["IMPORT_TOKEN_BUTTON"] = "import-token-button";
    ElementName["LANDING_PAGE_SWAP_ELEMENT"] = "landing-page-swap-element";
    ElementName["LEARN_MORE_LINK"] = "learn-more-link";
    ElementName["LEGACY_LANDING_PAGE_LINK"] = "legacy-landing-page-link";
    ElementName["MAX_TOKEN_AMOUNT_BUTTON"] = "max-token-amount-button";
    ElementName["NAVBAR_SEARCH_INPUT"] = "navbar-search-input";
    ElementName["NFT_ACTIVITY_TAB"] = "nft-activity-tab";
    ElementName["NFT_BUY_BAG_PAY_BUTTON"] = "nft-buy-bag-pay-button";
    ElementName["NFT_FILTER_BUTTON"] = "nft-filter-button";
    ElementName["NFT_FILTER_OPTION"] = "nft-filter-option";
    ElementName["NFT_TRENDING_ROW"] = "nft-trending-row";
    ElementName["PRICE_UPDATE_ACCEPT_BUTTON"] = "price-update-accept-button";
    ElementName["SUPPORT_LINK"] = "support-link";
    ElementName["SWAP_BUTTON"] = "swap-button";
    ElementName["SWAP_DETAILS_DROPDOWN"] = "swap-details-dropdown";
    ElementName["SWAP_TOKENS_REVERSE_ARROW_BUTTON"] = "swap-tokens-reverse-arrow-button";
    ElementName["TOKEN_SELECTOR_ROW"] = "token-selector-row";
    ElementName["TWITTER_LINK"] = "twitter-link";
    ElementName["WALLET_TYPE_OPTION"] = "wallet-type-option";
})(ElementName || (ElementName = {}));
