// @amplitude/analytics-client-common/lib/cjs/attribution/constants.js
Object.defineProperty(exports, "__esModule", { value: true });
exports.MKTG = exports.BASE_CAMPAIGN = exports.EMPTY_VALUE = exports.WBRAID = exports.TWCLID = exports.TTCLID = exports.MSCLKID = exports.KO_CLICK_ID = exports.GCLID = exports.GBRAID = exports.FBCLID = exports.DCLID = exports.UTM_CONTENT = exports.UTM_TERM = exports.UTM_CAMPAIGN = exports.UTM_MEDIUM = exports.UTM_SOURCE = void 0;
exports.UTM_SOURCE = 'utm_source';
exports.UTM_MEDIUM = 'utm_medium';
exports.UTM_CAMPAIGN = 'utm_campaign';
exports.UTM_TERM = 'utm_term';
exports.UTM_CONTENT = 'utm_content';
exports.DCLID = 'dclid';
exports.FBCLID = 'fbclid';
exports.GBRAID = 'gbraid';
exports.GCLID = 'gclid';
exports.KO_CLICK_ID = 'ko_click_id';
exports.MSCLKID = 'msclkid';
exports.TTCLID = 'ttclid';
exports.TWCLID = 'twclid';
exports.WBRAID = 'wbraid';
exports.EMPTY_VALUE = 'EMPTY';
exports.BASE_CAMPAIGN = {
    utm_source: undefined,
    utm_medium: undefined,
    utm_campaign: undefined,
    utm_term: undefined,
    utm_content: undefined,
    referrer: undefined,
    referring_domain: undefined,
    dclid: undefined,
    gbraid: undefined,
    gclid: undefined,
    fbclid: undefined,
    ko_click_id: undefined,
    msclkid: undefined,
    ttclid: undefined,
    twclid: undefined,
    wbraid: undefined,
};
exports.MKTG = 'MKTG';
