// @uniswap/analytics-events/lib/esm/properties.js
export var WalletConnectionResult;
(function (WalletConnectionResult) {
    WalletConnectionResult["FAILED"] = "Failed";
    WalletConnectionResult["SUCCEEDED"] = "Succeeded";
})(WalletConnectionResult || (WalletConnectionResult = {}));
export var SwapPriceUpdateUserResponse;
(function (SwapPriceUpdateUserResponse) {
    SwapPriceUpdateUserResponse["ACCEPTED"] = "Accepted";
    SwapPriceUpdateUserResponse["REJECTED"] = "Rejected";
})(SwapPriceUpdateUserResponse || (SwapPriceUpdateUserResponse = {}));
/** Known navbar search result types */
export var NavBarSearchTypes;
(function (NavBarSearchTypes) {
    NavBarSearchTypes["COLLECTION_SUGGESTION"] = "collection-suggestion";
    NavBarSearchTypes["COLLECTION_TRENDING"] = "collection-trending";
    NavBarSearchTypes["RECENT_SEARCH"] = "recent";
    NavBarSearchTypes["TOKEN_SUGGESTION"] = "token-suggestion";
    NavBarSearchTypes["TOKEN_TRENDING"] = "token-trending";
})(NavBarSearchTypes || (NavBarSearchTypes = {}));
/**
 * Known Filter Types for NFTs
 */
export var FilterTypes;
(function (FilterTypes) {
    FilterTypes["MARKETPLACE"] = "Marketplace";
    FilterTypes["PRICE_RANGE"] = "Price Range";
    FilterTypes["TRAIT"] = "Trait";
})(FilterTypes || (FilterTypes = {}));
