// @uniswap/analytics-events/lib/cjs/properties.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilterTypes = exports.NavBarSearchTypes = exports.SwapPriceUpdateUserResponse = exports.WalletConnectionResult = void 0;
var WalletConnectionResult;
(function (WalletConnectionResult) {
    WalletConnectionResult["FAILED"] = "Failed";
    WalletConnectionResult["SUCCEEDED"] = "Succeeded";
})(WalletConnectionResult = exports.WalletConnectionResult || (exports.WalletConnectionResult = {}));
var SwapPriceUpdateUserResponse;
(function (SwapPriceUpdateUserResponse) {
    SwapPriceUpdateUserResponse["ACCEPTED"] = "Accepted";
    SwapPriceUpdateUserResponse["REJECTED"] = "Rejected";
})(SwapPriceUpdateUserResponse = exports.SwapPriceUpdateUserResponse || (exports.SwapPriceUpdateUserResponse = {}));
/** Known navbar search result types */
var NavBarSearchTypes;
(function (NavBarSearchTypes) {
    NavBarSearchTypes["COLLECTION_SUGGESTION"] = "collection-suggestion";
    NavBarSearchTypes["COLLECTION_TRENDING"] = "collection-trending";
    NavBarSearchTypes["RECENT_SEARCH"] = "recent";
    NavBarSearchTypes["TOKEN_SUGGESTION"] = "token-suggestion";
    NavBarSearchTypes["TOKEN_TRENDING"] = "token-trending";
})(NavBarSearchTypes = exports.NavBarSearchTypes || (exports.NavBarSearchTypes = {}));
/**
 * Known Filter Types for NFTs
 */
var FilterTypes;
(function (FilterTypes) {
    FilterTypes["MARKETPLACE"] = "Marketplace";
    FilterTypes["PRICE_RANGE"] = "Price Range";
    FilterTypes["TRAIT"] = "Trait";
})(FilterTypes = exports.FilterTypes || (exports.FilterTypes = {}));
